extensions/oplogPopulator/OplogPopulator.js

```diff
@@ -198,27 +202,57 @@ class OplogPopulator {
     }
 
     /**
-     * start listening to bucket
+     * assigns a connector to a bucket
+     * @param {string} bucketName bucket name
+     * @returns {string} assigned connector for bucket
+     */
+     _assignConnectorForBucket(bucketName) {
+        const assignedConnector = this._connectorToBucket.get(bucketName);
+        // return previously assigned bucket if it exists
+        if (assignedConnector) {
+            return assignedConnector;
+        } else {
+            // else return the connector with the fewest buckets assinged to it
+            const bucketPerConnector = this._configuredConnectors.map(connector =>
```

Comment:
this algorithm will not scale, e.g. if we need to manage *many* connectors : we may need keep this list ordered somehow